Applications/SimpleShot/layers/centroid_nearest_neighbors.h

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2020 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   nearest_neighbors.h
+ * @date   09 Jan 2021
+ * @brief  This file contains the simple nearest neighbor layer, this layer
+ * takes centroid and calculate l2norm
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __NEAREST_NEIGHBORS_H__
+#define __NEAREST_NEIGHBORS_H__
+#include <string>
+
+/// @todo migrate these to API
+#include <layer_internal.h>
+#include <manager.h>
+
+#include <tensor.h>
+
+namespace simpleshot {
+namespace layers {
+
+/// @todo inherit this to API
+// class CentroidNearestNeighborsLayer : public ml::train::Layer {
+class CentroidNearestNeighborsLayer : public nntrainer::Layer {
```

Comment:
Nearest neighbor algorithm - https://en.wikipedia.org/wiki/Nearest_neighbour_algorithm
What this layer does is - https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm
Also, why centroid in the class name?

Reply:
Good point. 
This is not the K-nearest as well because it centroids, how about `CentroidKNearestNeighbors` ??

Reply:
Centroid can be a mode but the class should still be KNN?

Reply:
This is not generic KNN. IMO it's better to keep it as CentoridKNearestNeighbors, or better we can seperate centroid as a standalone layer (could do but takes manhour)

Later if we create a legit KNN, the layer or model can have KNN I guess.

Reply:
Renamed to `CentroidKNearestNeightbors` for now.

Reply:
I think centroid can go separate layer. and Use KNN classifier once we support KNN.

Reply:
Nearest neighbor needs a method to `identify the nearest neighbor` (it's eculidean distance in this) and `calculate the representative element for a group` (its centroid in this implementation). So, later KNN layer/model will have to support this as a config.

Reply:
Agreed